packages/relay/src/lib/eth.ts

```diff
@@ -1211,11 +1211,23 @@ export class EthImpl implements Eth {
       ]);
       if (result) {
         if (result?.type === constants.TYPE_TOKEN) {
+          const blockInfo = await this.getBlockInfo(blockNumber, requestDetails);
+          if (blockInfo) {
+            if (parseFloat(result.entity?.created_timestamp) > parseFloat(blockInfo.timestamp.to)) {
+              return EthImpl.emptyHex;
+            }
+          }
           if (this.logger.isLevelEnabled('trace')) {
             this.logger.trace(`${requestIdPrefix} Token redirect case, return redirectBytecode`);
           }
           return EthImpl.redirectBytecodeAddressReplace(address);
         } else if (result?.type === constants.TYPE_CONTRACT) {
+          const blockInfo = await this.getBlockInfo(blockNumber, requestDetails);
```

Comment:
we could move this bit right after the `if (result) {` given it's the same for both cases.